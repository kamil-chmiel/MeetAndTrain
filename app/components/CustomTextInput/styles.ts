import styled, { css } from 'styled-components/native';

export const Container = styled.View`
	height: 40;
	left: 0;
	right: 0;
	top: 50;
	align-items: flex-start;
	margin-bottom: 5;
	margin-top: 5;
`;

export const Title = styled.Text`
    text-align: left;
    color: gray;
    font-weight: bold;
`;

export const Input = styled.TextInput`
    flex: 1;
    width: 100%;
`;

export const Line = styled.View`
    height: 1;
    width: 100%;
    background-color: green;
`;