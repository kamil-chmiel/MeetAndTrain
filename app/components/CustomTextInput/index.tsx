import React from 'react';
import { Container, Input, Title, Line} from './styles';

const CustomTextInput = ({title}) => {
	return (
		<Container>
            <Title>{title}</Title>
			<Input/>
			<Line/>
		</Container>
	);
};

export default CustomTextInput;
