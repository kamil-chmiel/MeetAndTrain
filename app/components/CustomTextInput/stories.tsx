import React from 'react';
import { storiesOf } from '@storybook/react-native';

import CustomTextInput from '.';

storiesOf('Components/CustomTextInput', module).add('initial', () => <CustomTextInput />);
