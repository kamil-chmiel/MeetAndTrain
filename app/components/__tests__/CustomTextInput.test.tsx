import React from 'react'
import 'react-native'
import renderer from 'react-test-renderer'

import CustomTextInput from '../CustomTextInput'

const defaultProps = {
    title: "testTitle",
}

it('should render correctly', () => {
    const tree = renderer.create(<CustomTextInput {...defaultProps} />).toJSON()
    expect(tree).toMatchSnapshot()
})