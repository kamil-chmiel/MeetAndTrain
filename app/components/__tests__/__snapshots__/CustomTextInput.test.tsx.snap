// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "height": 40,
        "left": 0,
        "marginBottom": 5,
        "marginTop": 5,
        "right": 0,
        "top": 50,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "gray",
          "fontWeight": "bold",
          "textAlign": "left",
        },
      ]
    }
  >
    testTitle
  </Text>
  <TextInput
    allowFontScaling={true}
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "width": "100%",
        },
      ]
    }
    underlineColorAndroid="transparent"
  />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "green",
          "height": 1,
          "width": "100%",
        },
      ]
    }
  />
</View>
`;
